### ISSUE TITLE
[HOLD for payment 2025-01-15] [$125] Show pinned selfDM only to those users with "introSelected" for Track -- NVP:  newDotPersonalSpend

### ISSUE THREAD

**Issue by danielrvidal** [Expensify/App#53405]
As a follow up to[ this issue](https://github.com/Expensify/App/issues/52830), we're also going to remove the selfDM being pinned and in the LHN for all users going forward unless they have selected the Track case and have the Track intention NVP: `newDotPersonalSpend`

We currently show it to users who select this intention: 
- Get paid back from my employer case
- Split and chat with friends
- Something else

And we also pin it by default for users without an intention. We want to remove all these cases. 

So in summary, only users who select the Track intention should ever have the SelfDM pinned to their LHN by default when this issue is done. 

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021866165206858103237</li>
        <li>Upwork Job ID: 1866165206858103237</li>
        <li>Last Price Increase: 2024-12-10</li>
<li>Automatic offers: </li>
<ul>
<li>suneox | Reviewer | 105265880</li>
<li>nkdengineer | Contributor | 105320202</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @MonilBhavsar</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @nkdengineer</details>

**Comment by danielrvidal** [comment]
@suneox @nkdengineer since you all worked on [this one](https://github.com/Expensify/App/issues/52830), I assume you can also help complete this one quickly as a follow on? @MonilBhavsar can help if needed with anything.

**Comment by nkdengineer** [comment]
@danielrvidal I can help to check, please help to assign me here.

**Comment by danielrvidal** [comment]
Awesome, thank you!

**Comment by MonilBhavsar** [comment]
This is probably just one word change given we introduced the logic in the previous PR. Going to adjust the price to $125. @danielrvidal feel free to update it if you think otherwise.

**Comment by nkdengineer** [comment]
@danielrvidal So the expected here is the below right?

1. When the user signs in the first time, the selfDM will not be pinned by default.
2. If the user selects track NVP `newDotPersonalSpend`, the selfDM will be pinned.

Or 

1. When the user signs in the first time, the selfDM will be pinned by default.
2. If the user doesn't select `newDotPersonalSpend`, unpin the selfDM.

**Comment by MonilBhavsar** [comment]
I think it's the option two, but would let Daniel confirm

**Comment by danielrvidal** [comment]
It would be Option 1, sorry for the delay! We don't want them to know it exists unless they select the Track flow so we don't want it showing up before they select that.

**Comment by nkdengineer** [comment]
> It would be Option 1, sorry for the delay! We don't want them to know it exists unless they select the Track flow so we don't want it showing up before they select that.

@MonilBhavsar Based on this, we need to update the backend side to change `isPinned` of the selfDM to `false` when we create a new account and when we complete the onboarding flow with `newDotPersonalSpend` we update `isPinned` to `true`.

**Comment by MonilBhavsar** [comment]
Right! I'll work on it

**Comment by nkdengineer** [comment]
Great! @danielrvidal We must wait for the backend change before moving forward with the frontend PR.

**Comment by MonilBhavsar** [comment]
I think you can have a draft up, and then we can test it once backend is done?

**Comment by flaviadefaria** [comment]
Hey team, can we expect a PR for this today? This is a blocker for migrating cohort 1, which we plan to do next week!

**Comment by MonilBhavsar** [comment]
@danielrvidal @flaviadefaria I was thinking how does this work with migrating users? They won't see the guided setup modal. So should we keep self DM for them?

**Comment by MonilBhavsar** [comment]
Thank you! So to sum it up.

For any new/migrated users self DM won't be showed to the users.
When new users sign up on newDot, we display them onboarding modal and if they select "Track expenses", we would only display and pin self DM. Does that sound correct?

**Comment by danielrvidal** [comment]
Yes, that is correct!

**Comment by MonilBhavsar** [comment]
Perfect! I'll start working on this today

**Comment by nkdengineer** [comment]
@MonilBhavsar Do we need a frontend change here or not?

**Comment by MonilBhavsar** [comment]
Ah yes we do need it. I unassigned as the account was closed

**Comment by MonilBhavsar** [comment]
@nkdengineer what do you think of this comment https://github.com/Expensify/App/issues/53405#issuecomment-2536213377
Please let us know. This is top priority issue currently

**Comment by nkdengineer** [comment]
@MonilBhavsar I'm preparing the draft locally but we cannot test now because the selfDM is pinned by default.

**Comment by MonilBhavsar** [comment]
Agree! But would be good to keep it prepared, so we can merge when backend is deployed, mostly on Monday

**Comment by nkdengineer** [comment]
Created a draft [here](https://github.com/Expensify/App/pull/54098).

**Comment by nkdengineer** [comment]
Sure will do some unit tests soon.

**Comment by MonilBhavsar** [comment]
Thank you so much!

**Comment by MonilBhavsar** [comment]
@suneox @nkdengineer I was testing PR with backend change and looks like since there is no optimistic self DM report, the animation looks jarring. Please see LHN and self DM appearing suddenly. Any ideas to fix it?


https://github.com/user-attachments/assets/9305f20d-2e2f-4bf9-a9a3-25ade911be16

**Comment by MonilBhavsar** [comment]
Also, I have submitted backend change for review. Hoping to get it live on Monday

**Comment by nkdengineer** [comment]
@MonilBhavsar Your backend change will not return the selfDM report in `OpenApp` API or returns it with `isPinned: false`

**Comment by MonilBhavsar** [comment]
We're not creating by default so `OpenApp` does not return it. We do create when user selects personal spend option

**Comment by nkdengineer** [comment]
If so other users that don't select the personal spend option will never have the selfDM, right?

**Comment by MonilBhavsar** [comment]
Sounds good 👍 
Let's do it. Please update your PR

**Comment by nkdengineer** [comment]
@MonilBhavsar I updated the PR. Another note: I need another param is `selfDMCreatedReportActionID` to generate the created action for the selfDM report.

**Comment by nkdengineer** [comment]
@MonilBhavsar How about the backend PR progress?

**Comment by MonilBhavsar** [comment]
PR is being reviewed https://github.com/Expensify/Auth/pull/13540
I have to resolve conflicts and address comments

**Comment by nkdengineer** [comment]
@MonilBhavsar Is backend PR completed? I see we have this issue now https://github.com/Expensify/App/issues/54345.

**Comment by MonilBhavsar** [comment]
Thanks for linking!
I have decided to split the PR into two as there were lots of changes

1. The first part is deployed. where server does not create self DM everytime and only when user selects "Track expense" in onboarding flow. 
2. I am now working on second part, that will create self DM report with reportID and reportActionID provided by client to fix the flickering or self DM appearing after sometimes on newDot

**Comment by MonilBhavsar** [comment]
Submitted two more PR's for review

**Comment by nkdengineer** [comment]
@MonilBhavsar Thanks. Will retest and open the PR tomorrow.

**Comment by danielrvidal** [comment]
@MonilBhavsar do you think you could help us get this one out quickly? It would be great to have this out next week when we migrate.

**Comment by suneox** [comment]
# BugZero Checklist:

- [x] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [x] 1z. Other: New feature request.

Where bug was reported:
  - [ ] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [x] 2z. Other: New feature request.

Who reported the bug:
  - [ ] 3a. Expensify user
  - [x] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [x] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment: *N/A: This is a new feature request.*

- [x] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion: *N/A*

- [x] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

    *N/A: Due to it isn't an impactful bug*

**Comment by sakluger** [comment]
@nkdengineer yes we can pay $250 for this since the scope increased. I'm going to resend offers through Upwork since the original ones expired. Please let me know once you've accepted the new offer.

Summarizing payment on this issue:

Contributor: @nkdengineer $250, sent offer via Upwork: https://www.upwork.com/nx/wm/offer/105716353
Contributor+: @suneox $250, sent offer via Upwork: https://www.upwork.com/nx/wm/offer/105716355

**Comment by sakluger** [comment]
Paid out @nkdengineer's contract!

**Comment by MonilBhavsar** [comment]
We can close this then?

**Comment by suneox** [comment]
I've accepted the offer and waiting for payment

**Comment by sakluger** [comment]
Paid @suneox's contract. Now we can close :)
